project/schema_admin.py

```diff
@@ -0,0 +1,148 @@
+        logger.info("User must be authenticated!")
+        return False
+
+    if not user.is_staff:
+        logger.info("User must be staff!")
+        return False
+
+    if not is_request_user_verified(request):
+        logger.info("User must be verified via two-factor authentication!")
+        return False
+
+    if not user.has_perm(permission):
+        logger.info("User does not have permission to view text messages!")
+        return False
+
+    return True
+
+
+def ensure_request_has_verified_user_with_permission(permission: str):
+    def decorator(fn):
+        @wraps(fn)
+        def wrapper(parent, info: ResolveInfo, *args, **kwargs):
+            if not is_request_verified_user_with_permission(info.context, permission):
+                return None
+
+            return fn(parent, info, *args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
+def normalize_phone_number(phone_number: str) -> str:
+    """
+    Given either a 10-digit phone number or a U.S. phone number in E.164 format,
+    returns its 10-digit representation.
+
+    >>> normalize_phone_number('5551234567')
+    '5551234567'
+    >>> normalize_phone_number('+15551234567')
+    '5551234567'
+    """
+
+    if phone_number.startswith("+1"):
+        phone_number = phone_number[2:]
+    return phone_number
+
+
+@ensure_request_has_verified_user_with_permission(CHANGE_USER_PERMISSION)
+def resolve_user_admin_details(
+    parent, info, phone_number: Optional[str] = None, email: Optional[str] = None
+) -> Optional[JustfixUser]:
+    if phone_number:
+        phone_number = normalize_phone_number(phone_number)
+        return JustfixUser.objects.filter(phone_number=phone_number).first()
+    elif email:
+        return JustfixUser.objects.filter(email__iexact=email).first()
+    # TODO: Maybe raise some kind of error?
+    return None
+
+
+@ensure_request_has_verified_user_with_permission(CHANGE_USER_PERMISSION)
+def resolve_user_search(parent, info, query: str) -> List[JustfixUser]:
+    from users.admin import JustfixUserAdmin
+    from project.admin import JustfixAdminSite
+
+    MAX_USER_SEARCH_RESULTS = 10
+
+    if not query:
+        return []
+
+    users, has_duplicates = JustfixUserAdmin(JustfixUser, JustfixAdminSite()).get_search_results(
+        info.context, JustfixUser.objects.all(), query
+    )
+    return list(users[:MAX_USER_SEARCH_RESULTS])
+
+
+@schema_registry.register_queries
+class AdminQueries:
+    user_details = graphene.Field(
+        JustfixUserType,
+        phone_number=graphene.String(),
+        email=graphene.String(),
+        resolver=resolve_user_admin_details,
+    )
+
+    user_search = graphene.Field(
+        graphene.List(graphene.NonNull(JustfixUserType)),
+        query=graphene.String(required=True),
+        resolver=resolve_user_search,
+    )
+
+    is_verified_staff_user = graphene.Boolean(
+        resolver=ensure_request_has_verified_user_with_permission(CHANGE_USER_PERMISSION)(
```

Comment:
er technically it doesn't make sense to check for the `CHANGE_USER_PERMISSION` permission here, we should just be checking to see if they are a verified staff user, because that's what the field is called.